reference/src/layout/pointers.md

```diff
@@ -40,23 +39,23 @@ word and have therefore the same layout as C pointers.
 > or, in the case of `&mut T`, aliasing.
 
 We do not make any guarantees about the layout of
-multi-trait objects `&(dyn T + U)` or references to other dynamically sized types,
+multi-trait objects `&(dyn Trait1 + Trait2)` or references to other dynamically sized types,
 other than that they are at least word-aligned, and have size at least one word.
 
-The layout of `&dyn T` when `T` is a trait is the same as that of:
-```rust,ignore
+The layout of `&dyn Trait` when `Trait` is a trait is the same as that of:
+```rust
 #[repr(C)]
 struct DynObject {
-  data: *u8,
-  vtable: *u8,
+  data: *const u8,
```

Comment:
We deliberately did not have `mut` or `const` here because it kind of depends on whether this is used in a shared or mutable pointer. `const` unconditionally here is wrong.

Reply:
This is associated with `&dyn T`, so i think it's a shared pointer? or am i totally wrong about this?

By the way, does this mean `std::raw::TraitObject` (which only has a mut variant) is fundamentally broken?

And i feel the associated example in docs: ```data: &other_value as *const _ as *mut (),``` is even more broken https://doc.rust-lang.org/nightly/std/raw/struct.TraitObject.html

Reply:
@RalfJung the document only claims that the layout is equivalent to the layout of these structs, not that we internally use these structs to represent stuff.

I would prefer if these could be Rust code, and I wouldn't mind having two types here, `DynObject` and `DynObjectMut` if that's what it takes.

Reply:
> This is associated with &dyn T, so i think it's a shared pointer? or am i totally wrong about this?

This is, I think, meant to apply to `&dyn T` and `&mut dyn T` alike. It would be weird to have to repeat all that.

Fair. So maybe add a comment saying that `data` might be mutable if this is an `&mut dyn`?

Reply:
Done.